daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Transaction.scala

```diff
@@ -613,4 +604,176 @@ object Transaction {
       fa: FailedAuthorization,
   ) extends TransactionError
 
+  /** Whether `other` is the result of reinterpreting this transaction.
+    *
+    * @note This function is asymmetric.
+    */
+  def isReplayedBy[Nid, Cid, Val](
+      recorded: GenTransaction[Nid, Cid, Val],
+      replayed: GenTransaction[Nid, Cid, Val],
+  )(implicit ECid: Equal[Cid], EVal: Equal[Val]): Either[ReplayMismatch[Nid, Cid, Val], Unit] = {
+    import scalaz.syntax.equal._
+    import scalaz.std.option._
+
+    type Exe = Node.NodeExercises[Nid, Cid, Val]
+
+    def fail(recordedMismatch: Option[Nid], replayMismatch: Option[Nid]) =
+      Left(ReplayMismatch(recorded, recordedMismatch, replayed, replayMismatch))
+
+    @tailrec
+    def loop(
+        nids1: Stream[Nid],
+        nids2: Stream[Nid],
+        stack: List[(Nid, Exe, Stream[Nid], Nid, Exe, Stream[Nid])] = List.empty,
+    ): Either[ReplayMismatch[Nid, Cid, Val], Unit] =
+      (nids1, nids2) match {
+        case (nid1 #:: rest1, nid2 #:: rest2) =>
```

Comment:
Is `#::` for streams what `::` is for lists?

Reply:
Yes.